at_client/lib/src/client/local_secondary.dart

```diff
@@ -120,6 +120,9 @@ class LocalSecondary implements Secondary {
       if (builder.isCached) {
         llookupKey += 'cached:';
       }
+      if(builder.isPublic){
+        llookupKey += 'public:';
+      }
       if (builder.sharedWith != null) {
         llookupKey += '${AtUtils.formatAtSign(builder.sharedWith!)}:';
       }
```

Comment:
Are cached, public and sharedWith all mutually exclusive? If so, perhaps have some defensive code in here to check that at most *one* of these three booleans is true?

Reply:
public and sharedWith are mutually exclusive(not cached). However there can be a key without any of these(self keys and encryption keys).